### PR TITLE
fix(helm): match control-plane nodes by label presence when value is empty

### PR DESCRIPTION
**Pull request by lexfrei**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

- Bug fixes

## Problem

There is no Kubernetes convention for the value of `node-role.kubernetes.io/control-plane` label — different distributions set different values:

- **kubeadm, Talos, kind**: empty string `""`
- **k3s**: `"true"`

The Helm chart's node discovery (`kubeovn.nodeIPs`) and pod scheduling (`nodeSelector`) used strict value matching. When `MASTER_NODES_LABEL` was set to `node-role.kubernetes.io/control-plane` (without `=value`), `nodeSelector` required an exact empty-string match, which failed on k3s where the value is `"true"`. Additionally, `kubeovn.nodeIPs` used `index` on node labels without checking key presence first, causing false matches on nodes that don't have the label at all.

We hit this in production deploying [Cozystack](https://github.com/cozystack/cozystack) on multi-master k3s clusters — kube-ovn pods could not schedule on control-plane nodes, and OVN RAFT consensus failed (ref: cozystack/cozystack#2242).

## Changes

**kube-ovn chart:**
- `_helpers.tpl`: add `hasKey` check before accessing label value; handle trailing `=` and values containing `=` correctly via `rest`+`join`; add `kubeovn.masterNodeAffinity` helper
- Deployment templates (central, monitor, ic-controller): replace `nodeSelector` label matching with `nodeAffinity` using `Exists` operator (no value specified) or `In` operator (specific value), merged into the existing `affinity` block alongside `podAntiAffinity`
- Add helm-unittest tests (6 cases including edge cases)

**kube-ovn-v2 chart:**
- `_helpers.tpl`: same `hasKey` fix with nil-safe check (`toString` for YAML null handling); add `kubeovn.masterNodeRequired` helper with separate `nodeSelectorTerm` per label for OR semantics (matching `kubeovn.nodeIPs` and values.yaml documentation: "Any node that has **any** of these labels")
- Deployment templates: use `kubeovn.masterNodeRequired` via the existing `kube-ovn.affinities.nodeAffinity` helper
- Add helm-unittest tests (5 cases including multi-label OR semantics)

## Behavior

| MASTER_NODES_LABEL | Before | After |
| --- | --- | --- |
| `key` (no value) | nodeSelector `key: ""` — only matches empty value | nodeAffinity `Exists` — matches any value |
| `key=value` | nodeSelector `key: "value"` — strict match | nodeAffinity `In ["value"]` — strict match |
| `key=` (trailing =) | nodeSelector `key: ""` — matches empty | nodeAffinity `Exists` — matches any value |

## Which issue(s) this PR fixes

Discovered via cozystack/cozystack#2242

### CODE REVIEW

**Comment by lexfrei** [comment]
The `lint-test (ipv4)` failure is unrelated to this PR. The CI installs both `kube-ovn-v2` and `kube-ovn` charts into the same cluster, and the v1 chart fails because the `ovn` ServiceAccount already exists from v2:

```
Error: INSTALLATION FAILED: unable to continue with install: ServiceAccount "ovn" in namespace "kube-system" exists and cannot be imported into the current release
```

This is a pre-existing issue — other recent PRs with chart changes also fail on this workflow (e.g. `fix-no-primary-cni-svc-fail` with 3 consecutive failures).

All other checks (DCO, build, Mergify) pass.

File: charts/kube-ovn-v2/templates/_helpers.tpl [hunk 1]
```diff
@@ -64,7 +64,7 @@ a dry-run/template call and return nothing.
 {{- $ips := list -}}
 {{- range $node := $nodes.items -}}
   {{- range $label, $value := $.Values.masterNodesLabels }}
-  {{- if eq (index $node.metadata.labels $label) $value -}}
+  {{- if and (hasKey $node.metadata.labels $label) (or (eq ($value | toString) "") (eq (index $node.metadata.labels $label) $value)) -}}
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply charts/kube-ovn-v2/templates/_helpers.tpl]: Good catch. Fixed in 6dcf2b4 — added `toString` to both the value comparison in `kubeovn.nodeIPs` and the `values` list construction in `kubeovn.masterNodeRequired`. Added a test case that verifies unquoted YAML `true` (boolean) is correctly coerced to string `"true"` in the `In` operator.

File: charts/kube-ovn-v2/templates/_helpers.tpl [hunk 2]
```diff
@@ -81,6 +81,24 @@ a dry-run/template call and return nothing.
...
+(matching the kubeovn.nodeIPs helper which also uses OR).
+Uses Exists operator for empty/nil-value labels and In for specific values.
+*/}}
+{{- define "kubeovn.masterNodeRequired" -}}
+{{- $terms := list -}}
+{{- range $key, $value := .Values.masterNodesLabels -}}
+  {{- if eq ($value | toString) "" -}}
+    {{- $terms = append $terms (dict "matchExpressions" (list (dict "key" $key "operator" "Exists"))) -}}
+  {{- else -}}
+    {{- $terms = append $terms (dict "matchExpressions" (list (dict "key" $key "operator" "In" "values" (list $value)))) -}}
```
[the remark this answers is not in the document]
Comment (lexfrei) [reply charts/kube-ovn-v2/templates/_helpers.tpl]: Fixed in the same commit (6dcf2b4) — `$value | toString` applied to the `values` list in `kubeovn.masterNodeRequired`. Test added.

**Comment by oilbeater** [comment]
Thanks! @lexfrei
